api_service.py

```
@@ -1563,20 +1733,34 @@ async def oauth_callback(
-        )
-        with urllib.request.urlopen(userinfo_request, timeout=10) as response:
-            userinfo = json.loads(response.read().decode('utf-8'))
-            email_address = userinfo.get('email')
+        try:
+            userinfo_request = urllib.request.Request(
+                "https://www.googleapis.com/oauth2/v2/userinfo",
+                headers={"Authorization": f"Bearer {access_token}"}
+            )
+            with urllib.request.urlopen(
+                userinfo_request, timeout=10
+            ) as response:
```

Ruff (0.14.10)

Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)